Plugins/MatrixPlugin/MatrixPlugin.cs

```diff
@@ -0,0 +1,149 @@
+using Chummer;
+using Chummer.Plugins;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using NLog;
+using Microsoft.ApplicationInsights.Channel;
+using System.Diagnostics;
+using System.Xml.XPath;
+using System.Xml.Serialization;
+using System.Xml;
+
+namespace MatrixPlugin
+{
+    [DebuggerDisplay("{" + nameof(GetDebuggerDisplay) + "(),nq}")]
+    public class MatrixPlugin : IPlugin
+    {
+
+        private static readonly Logger _log = LogManager.GetCurrentClassLogger();
+        public List<MatrixAction> ActionList;
+        public List<Program> ProgramList;
+
+        public override string ToString()
+        {
+            return "Matrix Helper";
+        }
+
+        public void CustomInitialize(frmChummerMain mainControl)
+        {
+            try
+            {
+                ActionList = new List<MatrixAction>();
+                XmlSerializer xmlSerializerAction = new XmlSerializer(typeof(MatrixAction));
+                XPathNavigator navActions = XmlManager.LoadXPath("actions.xml");
```

Comment:
I mentioned this in my last review, but you really need to be careful about doing this sort of caching for stuff that would be used for specific characters. In Chummer right now, every single character could potentially have its own, different set of custom datasets loaded, and if any of these datasets modify the file you're using, then there is going to be a mismatch between what the user expects how their character should behave and how it will actually behave. For example, if I am using a custom dataset that amends `actions.xml` such that Control Device is a Hacking + LOG action by default instead of EWar + INT, this code will still load Control Device as EWar + INT for my character, which is not good.